### DISCUSSION TITLE
Replace Task with ValueTask

### DISCUSSION THREAD

**Discussion by vboltsis** [dotnetcore/EasyCaching#412] [Ideas]
Hello, I wanted to ask if an implementation of all methods with ValueTask instead of Task is in the roadmap to be supported since the library is meant for caching.
I am willing to create a PR with the necessary changes but I wanted to know if there is  some aspect of the implementation I did not consider that would break something.
Thanks!

USEFUL LINKS:
https://blog.stephencleary.com/2020/03/valuetask.html
https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.valuetask-1?view=net-6.0
https://www.youtube.com/watch?v=dCj7-KvaIJ0

**Comment by catcherwong** [comment]
What are the benefits of changing  Task to  ValueTask?

**Reply by vboltsis** [reply]
If a method does not make an io call but gets the value from memory, valuetask uses less memory (it is a strict) and it is faster. In the useful links, watch the video.

**Reply by catcherwong** [reply]
Got it. Welcome to submit a PR! 😄
